### PR TITLE
[Tables] Add __repr__ and __str__ methods to models

### LINKED ISSUE #19278: [Tables] Add __repr__ methods to custom models
This will help the debugging experience

### PR DESCRIPTION
**Pull request by seankane-msft**
closes #19278 

Will help make debugging easier for us and customers

### CODE REVIEW

File: sdk/tables/azure-data-tables/azure/data/tables/_models.py [hunk 1]
```diff
@@ -191,6 +221,14 @@ def _from_generated(cls, generated, **kwargs):  # pylint: disable=unused-argumen
             days=generated.days,
         )
 
+    def __str__(self):
+        # type: () -> str
+        return u"enabled={}, days={}".format(self.enabled, self.days)
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/azure/data/tables/_models.py:226]: `__str__` is not supposed to return unicode. Must be type `str`

Comment (annatisch) [reply sdk/tables/azure-data-tables/azure/data/tables/_models.py:226]: Not sure whether the same applied to `__repr__` - would need to check the Py2.7 stdlib ref docs.

File: sdk/tables/azure-data-tables/azure/data/tables/_models.py [hunk 2]
```diff
@@ -617,8 +682,13 @@ def __init__(self, **kwargs):
         )
 
     def __str__(self):
+        # type: () -> str
         return self._str
 
+    def __repr__(self):
+        # type: () -> str
+        return u"AccountSasPermissions(_str={})".format(self._str)
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/azure/data/tables/_models.py:690]: I wouldn't document `_str` - this isn't supposed to be public.
I would probably either document it as just the string:
`return "AccountSasPermissions({})".format(self._str)`
Or you could go fully verbose:
`return "AccountSasPermissions(read={}, write={}, ...)".format(self.read, self.write, ...)`

File: sdk/tables/azure-data-tables/azure/data/tables/_models.py [hunk 3]
```diff
@@ -546,8 +606,13 @@ def __init__(self, **kwargs):
         self._str = ("s" if self.service else "") + ("o" if self.object else "")
 
     def __str__(self):
+        # type: () -> str
         return self._str
 
+    def __repr__(self):
+        # type: () -> str
+        return u"<ResourceTypes(_str={})".format(self.__str__())
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/azure/data/tables/_models.py:614]: Sneaky angle bracket at the front there.

File: sdk/tables/azure-data-tables/azure/data/tables/_models.py [hunk 4]
```diff
@@ -153,6 +173,16 @@ def _from_generated(cls, generated):
             )
         )
 
+    def __str__(self):
+        # type: () -> str
+        return "version={}, enabled={}, include_apis={}, retention_policy={}".format(
+            self.version, self.enabled, self.include_apis, self.retention_policy
+        )
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/azure/data/tables/_models.py:180]: I'm thinking maybe we just don't define `__str__` for the time being. Formatting with an instance will default to using `__repr__` in the meantime anyway.

File: sdk/tables/azure-data-tables/azure/data/tables/_models.py [hunk 5]
```diff
@@ -153,6 +173,16 @@ def _from_generated(cls, generated):
...
+
+    def __repr__(self):
+        # type: () -> str
+        return "TableMetrics({})".format(self.__str__())
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/azure/data/tables/_models.py:184]: Even though it's pretty safe to assume we'll never hit it for these models, we tend to add `[:1024]` to the end.

File: sdk/tables/azure-data-tables/azure/data/tables/_models.py (hunk 2 above)
Comment (seankane-msft) [reply sdk/tables/azure-data-tables/azure/data/tables/_models.py:690]: Went with the latter for consistency

File: sdk/tables/azure-data-tables/azure/data/tables/_models.py (hunk 4 above)
Comment (seankane-msft) [reply sdk/tables/azure-data-tables/azure/data/tables/_models.py]: I went with this suggestion and removed `__str__` methods for now, we can always add them back later.

**Review by iscai-msft — APPROVED** [review]

**Pull request merged**